### ISSUE TITLE
Hosts with bad IP

### ISSUE THREAD

**Issue by cedvan** [rancher/rancher#2946]
Hi,

I need your help because I don't understand problem.

#### First

I running `rancher-server` on host debian 8 with local IP `192.168.1.10` : 

```
sudo docker run \
        -d \
        --restart=always \
        --name rancher-server \
        -e "VIRTUAL_HOST=infra.mydomain.com" \
        -e "VIRTUAL_PORT=8080" \
        -e "VIRTUAL_PROTO=http" \
        -v /data/infra/data/rancher-server/mysql:/var/lib/mysql \
        rancher/server
```
*`VIRTUAL_` variables present to configure [Docker Nginx Proxy](https://github.com/jwilder/nginx-proxy)*

#### Second

On same host, I running 3 VM Rancher OS with vagrant : 

- rancher-node-01 : local IP  `192.168.2.1`.
- rancher-node-02 : local IP  `192.168.2.2`.
- rancher-node-03 : local IP  `192.168.2.3`.

On each VM, I run a `rancher-agent` register to `rancher-server` with this command :

```
sudo docker run \
        -d \
        --privileged \
        --add-host infra.mydomain.com:192.168.1.10 \
        -v /var/run/docker.sock:/var/run/docker.sock \
        rancher/agent:v0.8.2 \
        https://infra.mydomain.com/v1/scripts/.....
```

#### Result

In rancher UI : 

![rancher-hosts](https://cloud.githubusercontent.com/assets/4289595/11661131/a3ea9586-9dd0-11e5-9fe9-6d2edc652f3a.png)

The hosts use the IP `192.168.1.10` of the main host instead of configure itself with VM IP, and my services containers blocked to Networking. This two problems are linked I suppose?

Please need your help :)

Thank you,
Cedvan

**Comment by cloudnautique** [comment]
@cedvan, your going through a NAT I take it? In order for the IPs to register into the nodes you need to add `-e CATTLE_AGENT_IP=<ip of node>` to the rancher-agent registration command.

**Comment by cedvan** [comment]
Ok, IP are good now, thanks !

My service are always blocked to Networking... 
Is it necessary to manually open the udp ports 500 and 4500 on RancherOS , or is it automatic?

Monitoring of rancher-agent from `rancher-node-01` blocked to `connecting...` normal ?

![rancher-node-01](https://cloud.githubusercontent.com/assets/4289595/11663000/0e837d28-9dda-11e5-942e-2cb7d10b81d1.png)

**Comment by deniseschannon** [comment]
There's nothing you need to do in RancherOS to open the ports. 

What version of Rancher are you running? Have you looked at the troubleshooting docs to see if that might help? 

http://docs.rancher.com/rancher/faqs/troubleshooting/

**Comment by deniseschannon** [comment]
@cedvan I wanted to clarify that RancherOS doesn't block the ports, but where you launch RancherOS might have it blocked. For example, AWS would need updating of your security groups. 

What version of Rancher are you using? This was recently fixed in v0.49.0. 
https://github.com/rancher/rancher/issues/2510

**Comment by cedvan** [comment]
@deniseschannon I resolved my problem ;)
